traffic.py

```diff
@@ -63,18 +62,23 @@ def send_waku_msg(node_address, kbytes, pubsub_topic, content_topic):
   print("Injecting traffic to single node REST API:", args.single_node)
 
 # this simply converts from http://url_[1..5]:port to
-# [http://url_1:port
+# [http://url_1:port or from http://url-[1..5]:port to
+# [http://url-1:port
 nodes = []
 if args.multiple_nodes:
-  range_nodes = args.multiple_nodes.split(":")[1].split("_")[2]
-  node_placeholder = args.multiple_nodes.replace(range_nodes, "{placeholder}")
-  clean_range = range_nodes.replace("[", "").replace("]", "")
-  start = int(clean_range.split("..")[0])
-  end = int(clean_range.split("..")[1])
+  match = re.search(r'\[(\d+)\.\.(\d+)\]', args.multiple_nodes)
+  if match:
+      start = int(match.group(1))
+      end = int(match.group(2))
+  else:
+      print("Could not parse range of multiple_nodes argument")
+      exit
 
   print("Injecting traffic to multiple nodes REST APIs") 
   for i in range(start, end+1):
-    nodes.append(node_placeholder.replace("{placeholder}", str(i)))
+      range_start = args.multiple_nodes.index("[")
```

Comment:
mind ellaborating why redefining here `range_start` and `range_end`. Isn't this the same as the `start` and `end` we got before?

isnt't the previous code ok?
```
nodes.append(node_placeholder.replace("{placeholder}", str(i)))
```

Reply:
I thought that, that would require unnecessary additional lines of code.
The line replacing '{placeholder}' in the for loop is dependent on this earlier line:
`node_placeholder = args.multiple_nodes.replace(range_nodes, "{placeholder}")`,
which is dependent on `range_nodes` being defined.
With the new method of determining start and end values, no `range_nodes` is defined and it seemed like unnecessary extra code to also go define it, just to then be able to use two other lines of code, so I did what seemed to be the most direct string replacement method.

Thinking it through now, I see that it ends up being about the same number of lines of code if I do something like this:

```
if args.multiple_nodes:
  range_nodes = "[" + args.multiple_nodes.split(":")[1].split("[")[2]
  node_placeholder = args.multiple_nodes.replace(range_nodes, "{placeholder}")
  match = re.search(r'\[(\d+)\.\.(\d+)\]', args.multiple_nodes)
  if match:
      start = int(match.group(1))
      end = int(match.group(2))
  else:
      print("Could not parse range of multiple_nodes argument")
      exit
      
  print("Injecting traffic to multiple nodes REST APIs") 
  for i in range(start, end+1):
    nodes.append(node_placeholder.replace("{placeholder}", str(i)))
```

I need to verify that it works and will then commit the change.
I'll also see if there is a better approach, please feel free to share suggestions if you spot a more efficient way of doing it.

Reply:
may be code golfing, but how about somethig like this? iirc this is what we are trying to do.

```python
import re

url = "http://waku-simulator-nwaku-[1..10]:8645"
start, end = (int(x) for x in re.search(r"\[(\d+)\.\.(\d+)\]", url).groups())

for i in range(end, start - 1, -1):
    print(re.sub(r"\[\d+\.\.\d+\]", str(i), url))
```

Reply:
That works, implemented in https://github.com/alrevuelta/rest-traffic/pull/8/commits/a3742fa18c678253aa2a92ed2e14585ad8dc509d